reporter/symbol_uploader.go

```diff
@@ -123,14 +124,22 @@ func NewDatadogSymbolUploader(cfg *SymbolUploaderConfig) (*DatadogSymbolUploader
 
 	compressDebugSections := !cfg.DisableDebugSectionCompression && CheckObjcopyZstdSupport()
 
+	httpClient := &http.Client{Timeout: uploadTimeout}
+	httpTransport, ok := http.DefaultTransport.(*http.Transport)
```

Comment:
This works in tests because `httpmock` overwrites `http.DefaultTransport` in https://github.com/jarcoal/httpmock/blob/v1.4.0/transport.go#L1124-L1140, so this will return `nil, false` and as such we will not override the mock transport expected in tests.